test/unit/alphabet/nucleotide/nucleotide_test_template.hpp

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2006-2018, Knut Reinert & Freie Universitaet Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI Molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE
+
+#include <sstream>
+
+#include <gtest/gtest.h>
+
+#include <range/v3/view/zip.hpp>
+
+#include <seqan3/alphabet/concept.hpp>
+#include <seqan3/alphabet/nucleotide/all.hpp>
+
+using namespace seqan3;
+
+template <typename T>
+class nucleotide : public ::testing::Test
+{};
+
+TYPED_TEST_CASE_P(nucleotide);
+
+TYPED_TEST_P(nucleotide, global_assign_char)
+{
+    using t = TypeParam;
+    std::vector<char> input
+    {
+        'A', 'C', 'G', 'T', 'U', 'N',
+        'a', 'c', 'g', 't', 'u', 'n',
+        'R', 'Y', 'S', 'W', 'K', 'M', 'B', 'D', 'H', 'V',
+        'r', 'y', 's', 'w', 'k', 'm', 'b', 'd', 'h', 'v',
+        '!'
+    };
+
+    std::vector<TypeParam> cmp;
+    if constexpr (std::is_same_v<TypeParam, dna4> || std::is_same_v<TypeParam, rna4>)
```

Comment:
remove all those specific tests and put them into the corresponding dna4/... test files. Same for `to_char`

Instead I would first only test for `A,C,G` with to_char and from_char as they are common between all alphabets,
and then
```cpp
for (t::rank_type rnk = 0; rnk < alphabet_size_v<t>; rnk++)
{
    t tmp{};
    assign_rank(tmp, rnk); // assuming assign_rank and to_rank are tested with the alphabet test beforehand
    from_char(tmp, to_char(rnk))); 
    EXPECT_EQ(rnk, to_rank(tmp));
}
``` 
to test that the two functions are consistent.
What do you think? Same could be done for amino acids?